avid-adventurers/src/pages/survey/Commonality.tsx

```diff
@@ -1,26 +1,81 @@
-import React from 'react';
-import { useNavigate } from 'react-router-dom';
+import React, { useState, useEffect } from 'react';
+import { useNavigate, useLocation } from 'react-router-dom';
 import { Label } from '../../components/Label/Label';
-import { Container, FormWrapper, ButtonRow, ContinueButton } from '../styles';
+import { Container, FormWrapper, ButtonRow, ContinueButton, BackButton } from '../styles';
 import { useSurvey } from '../../utils/surveyContext';
+import { TagContainer, Tag, RemoveButton } from '../../components/TagSelector/styles';
+import ProgressBar from '../../components/ProgressBar/ProgressBar';
 
 
 const Commonality: React.FC = () => {
     const navigate = useNavigate();
+    const location = useLocation();
     const { friendProfileImage, friendName, friendInterests, activity, interests, rating, setRating, updateInterests, setProfileImage } = useSurvey();
 
+    const [selectedTags, setSelectedTags] = useState<string[]>(() => {
```

Comment:
you should not need to use localStorage and useContext like this with each other! there's a lot of cleanup we can do here with all the redundant calls to localStorage and useEffect.